src/Sarif/Core/Run.cs

```diff
@@ -13,6 +14,109 @@ public partial class Run
         private static Invocation EmptyInvocation = new Invocation();
         private static LogicalLocation EmptyLogicalLocation = new LogicalLocation();
 
+        private IDictionary<FileLocation, int> _fileToIndexMap;
+
+        public int GetFileIndex(
+            FileLocation fileLocation, 
+            bool addToFilesTableIfNotPresent = true,
+            OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
+            System.Text.Encoding encoding = null)
+        {
+            if (fileLocation == null) throw new ArgumentNullException(nameof(fileLocation));
+
+            if (this.Files == null || this.Files.Count == 0)
```

Comment:
>this.Files == null || this.Files.Count == 0 [](start = 16, length = 43)

`this.Files?.Any()`... BUT! If `Files` is missing and `addToFilesTableIfNotPresent` is `true`, then wouldn't this be a good time to create `Files`?

Reply:
Well, I guess I see what you're after -- if the producer doesn't initialize `run.Files` with the list of scanned files, maybe that means the producer didn't _want_ a files array, so we shouldn't force it on him here. But in that case, maybe the producer just shouldn't call this method?

---
In reply to: [244521702](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521702) [](ancestors = 244521702)

Reply:
No. This is just a bad bug you've identified that the unit test authoring has closed. In the case of files being null and 'addFileToTableIfNotPresent' == true, that boolean value would not be honored. Basically, this method should never return -1 if that value is true

---
In reply to: [244521733](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521733) [](ancestors = 244521733,244521702)